app/questionnaire/placeholder_parser.py

```diff
@@ -179,18 +187,39 @@ def _resolve_value_source_list(
         return values
 
     def _get_routing_path_block_ids(
-        self, data: Sequence[Mapping], sections_to_ignore: list | None = None
+        self,
+        *,
+        data: Sequence[Mapping],
+        assess_routing_path: bool,
+        sections_to_ignore: list | None = None,
     ) -> dict[tuple, tuple[str, ...]] | None:
-        if self._location:
-            return get_block_ids_for_calculated_summary_dependencies(
-                schema=self._schema,
-                location=self._location,
-                progress_store=self._progress_store,
-                sections_to_ignore=sections_to_ignore,
-                data=data,
-                path_finder=self._path_finder,
+        if not self._location:
+            return {}
+
+        dependent_sections: dict[str, set[str]] | dict[str, OrderedSet[str]]
+        if assess_routing_path:
+            dependent_sections = self._schema.placeholder_section_dependencies_by_block[
+                self._location.section_id
+            ]
+            source_type = "answers"
+        else:
+            dependent_sections = (
+                self._schema.calculated_summary_section_dependencies_by_block[
+                    self._location.section_id
+                ]
             )
-        return {}
+            source_type = "calculated_summary"
```

Comment:
Why do we have `assess_routing_path` here?. Currently it could be true if any of the placeholders in the list has a transform that needs routing path. But that list could also contain a calculated summary placeholder which would then break?. As it would be True, for a calc summary,

Reply:
Removed 👍